Snippets/Sqs/Sqs_All/QueueCreation/QueueCreationUtils.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace SqsAll.QueueCreation
+{
+    using System;
+    using System.Threading.Tasks;
+    using Amazon.SQS;
+    using Amazon.SQS.Model;
+    using Sqs_All;
+
+    #region sqs-create-queues
+
+    public static class QueueCreationUtils
+    {
+        static TimeSpan DefaultTimeToLive = TimeSpan.FromDays(4);
+
+        public static async Task CreateQueue(string queueName, TimeSpan? maxTimeToLive = null)
+        {
+            try
+            {
+                using (var client = ClientFactory.CreateSqsClient())
+                {
+                    var sqsRequest = new CreateQueueRequest
+                    {
+                        QueueName = QueueNameHelper.GetSqsQueueName(queueName)
+                    };
+                    var value = Convert.ToInt32((maxTimeToLive ?? DefaultTimeToLive).TotalSeconds).ToString();
+                    sqsRequest.Attributes.Add(QueueAttributeName.MessageRetentionPeriod, value);
+                    await client.CreateQueueAsync(sqsRequest)
+                        .ConfigureAwait(false);
+                }
+            }
+            catch (QueueNameExistsException)
```

Comment:
I'm not sure about truncation - I added that to the transport as an internal-only configuration option solely to get the ATs working, I'm not sure if it's a good way to handle real systems. To date I have just been relying on users choosing endpoint names that are short enough - this gives a very straightforward experience when a user goes looking for their queue in the SQS portal. I'm a little concerned that the pretruncation will be more confusing for new users and will give a worse experience compared to just getting an exception saying "queue name is too long", but I'm open to suggestions here.

In any case I don't think we should add the pretruncation support here because it's not possible for users to configure it in their endpoints at this time.